src/player_hardcoded_effects.cpp

```diff
@@ -1110,6 +1111,45 @@ void player::hardcoded_effects( effect &it )
                     }
                 }
             }
+            if( ( ( has_trait( trait_id( "SCHIZOPHRENIC" ) ) || has_artifact_with( AEP_SCHIZO ) ) && one_in( 7200 ) && is_player() ) ) {
+                if( one_in( 2 ) ) {
+                    sound_hallu();
+                } else {
+                    int x = posx();
+                    int y = posy();
+                    int z = posz();
+
+                    tripoint mp = tripoint( x, y, z );
+                    
+                    int max_count = rng( 1, 3 );
+                    int count = 0;
+                    bool found_valid = false;
+                    for( int i = -1; i <= 1; i++ ) {
```

Comment:
How about this much shorter code:
```C++
for( const tripoint &mp : g->m.points_in_radius( pos(), 1 ) ) {
    if( mp == pos() ) {
        continue;
    }
    // use mp to check the terrain and do stuff
}
```
`map::points_in_radius` returns a list of points around the given one (the center of a rectangle), with the second parameter being the radius. The call above will yield the same points that your loops iterate over.